### ISSUE TITLE
Tests: Remove redundant ThemeProvider from tests

### ISSUE THREAD

**Issue by siddharthkp** [primer/react#4514]
There are multiple tests that wrap the rendered component with ThemeProvider: https://github.com/search?q=repo%3Aprimer%2Freact+%3CThemeProvider+path%3Atest.tsx&type=code

We do not have to do this anymore because [we already wrap all stories in a ThemeProvider in preview.js](https://github.com/primer/react/blob/f9eae58360ebaac9f3ada777e11f954d3013782a/packages/react/.storybook/preview.js#L222). 

Note: There might be a couple exceptions so make sure to test this by switching themes in storybook

**Comment by lesliecdubs** [comment]
👋 What is the impact of the redundant ThemeProvider in tests? Is this a "nice-to-have" quality of life improvement?

**Comment by siddharthkp** [comment]
> Is this a "nice-to-have" quality of life improvement?

Yep, I'd say this is nice to have. I don't think there are any bugs that are missed because of this redundancy

**Comment by francinelucca** [comment]
Was able to remove some over in https://github.com/primer/react/pull/6224 but the others were erroring out without it, so the ThemeProvider is actually needed in some cases
